### PR TITLE
updated main.rkt to support selective generation of rosette's output-…

### PR DESCRIPTION
**Pull request by cknizek**
…smt feature from the commandline

(PR in response to [@gussmith23's issue](https://github.com/uwsampl/lakeroad/issues/453))

### What this change does:
This change modifies `main.rkt` to support the command line flag `--output-smt-path "path"`. The specified path will be used as the output to Rosette's `output-smt` feature.

The path you specify will become a folder, as there may be many SMT output files generated and placed in the selected path.

### How this change does it:
It adds a new parameter, `output-smt-path` (initially set to `#f`) to `main.rkt`.

If `output-smt-path` becomes no longer equal to `#f` (i.e. it has been assigned in the commandline arguments to become something), the SMT output feature of Rosette will be activated. The output will be directed to the path specified.

### How to use this change (see line 2 of the below code block):
```
// RUN: racket $LAKEROAD_DIR/bin/main.rkt \
// RUN:  --output-smt-path "example_test_using_include_smt_output_test" \
// RUN:  --solver bitwuzla \
// RUN:  --verilog-module-filepath %s \
// RUN:  --architecture xilinx-ultrascale-plus \
// RUN:  --template dsp \
// RUN:  --out-format verilog \
// RUN:  --top-module-name top \
// RUN:  --verilog-module-out-signal out:11 \
// RUN:  --pipeline-depth 1 \
// RUN:  --clock-name clk \
// RUN:  --module-name out \
// RUN:  --input-signal a:11 \
// RUN:  --input-signal b:11 \
// RUN:  --input-signal c:11 \
// RUN:  --input-signal d:11 \
// RUN:  --extra-cycles 3 \
// RUN:  --timeout 120 \
// RUN: | FileCheck %s
```

### CODE REVIEW

File: bin/main.rkt [hunk 1]
```diff
@@ -144,6 +145,7 @@
       (error "Output signal must be specified as <name>:<bw>"))
     (verilog-module-out-signal (first splits))
     (verilog-module-out-bitwidth (string->number (second splits))))]
+ ["--output-smt-path" v "Specify the output of the SMT solver to a file." (output-smt-path v)]
```
Comment (gussmith23) [inline bin/main.rkt:148]: This path actually refers to the _directory_ where the files will be placed, not the file

File: bin/main.rkt [hunk 2]
```diff
@@ -538,4 +544,4 @@
                                   (error "Yosys failed."))))
        (output-port))]
 
-     [_ (error "Invalid output format.")])])
+     [_ (error "Invalid output format.")])])
```
Comment (gussmith23) [inline bin/main.rkt:547]: can you undo this change? i'm ot sure why your editor keeps making it

**Review by gussmith23 — COMMENTED** [review]
Wonderful!!!! Great initiative on adding this!!!

File: bin/main.rkt (hunk 2 above)
Comment (cknizek) [reply bin/main.rkt:547]: > can you undo this change? i'm ot sure why your editor keeps making it

Yeah I can do undo it. I think this happened after I ran `raco fmt` on the file.

File: bin/main.rkt (hunk 1 above)
Comment (cknizek) [reply bin/main.rkt:148]: Oops yeah you're right, I'll change the line to reflect this

**Review by gussmith23 — CHANGES_REQUESTED** [review]
Looks like you accidentally deleted main.rkt...

**Comment by cknizek** [comment]
> Looks like you accidentally deleted main.rkt...

Dang, thanks for checking this... I think this somehow happened when I was trying to pipe the `raco fmt bin/main.rkt` output directly into `bin/main.rkt`.

I think the issue should be resolved now

File: bin/main.rkt [hunk 3]
```diff
@@ -128,6 +129,7 @@
     (when (not (file-exists? v))
       (error (format "File ~a does not exist." v)))
     (verilog-module-filepath v))]
+ ["--output-smt-path" v "Specify the output of the SMT solver to a file." (output-smt-path v)]
```
Comment (gussmith23) [inline bin/main.rkt:132]: Again, this is a directory, not a file. Please change the docstring here.

**Review by gussmith23 — CHANGES_REQUESTED** [review]

File: bin/main.rkt (hunk 1 above)
Comment (gussmith23) [reply bin/main.rkt]: See this thread, still unresolved

**Review by gussmith23 — APPROVED** [review]
When CI goes through, go ahead and merge!
